### PR TITLE
adding getting alerts via alertmanager, which allows user to be able …

### PR DESCRIPTION
**Pull request by rmartinez3**
Hello,

Will like to add getting alerts via alertmanager, in which a user is able to silence alerts via alertmanager, when a reboot is occurring. So kured wont be blocked on active alerts. Let me know what is needed for this to be merged.

Thanks

### CODE REVIEW

File: cmd/kured/main.go [hunk 1]
```diff
@@ -141,6 +144,20 @@ func rebootBlocked() bool {
 			log.Warnf("Reboot blocked: %d active alerts: %v", count, alertNames)
 			return true
 		}
+	} else if alertmanagerURL != "" {
+		alertNames, err := alerts.AlertmanagerActiveAlerts(alertmanagerURL, alertFilter)
+		if err != nil {
+			log.Warnf("Reboot blocked: AlertManager query error: %v", err)
+			return true
+		}
+		count := len(alertNames)
```
Comment (bboreham) [inline cmd/kured/main.go]: Do you think this section could be merged with the one above where the code is identical?

File: pkg/alerts/alertmanager.go [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+
+	if err != nil {
+		return nil, err
+	}
+
+	activeAlertSet := make(map[string]bool)
+
+	for _, alert := range fetchedAlerts {
+		alertName := alert.Labels["alertname"]
+		if filter == nil || !filter.MatchString(string(alertName)) {
```
Comment (bboreham) [inline pkg/alerts/alertmanager.go]: From here to the end of the function seems identical to the prometheus query code - could it be extracted into a common function?

Comment (rmartinez3) [reply pkg/alerts/alertmanager.go]: I'll work on making a common setction for both of them . thanks for looking over PR .

**Comment by evrardjp** [comment]
Does this need to exist? Couldn't a generic notification command do the trick?

**Pull request closed without merging**